icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -3242,12 +3249,7 @@ int32_t SimpleDateFormat::subParse(const UnicodeString& text, int32_t& start, ch
 
         // Delayed checking for adjustment of Hebrew month numbers in non-leap years.
         if (saveHebrewMonth >= 0) {
-            HebrewCalendar *hc = (HebrewCalendar*)&cal;
```

Comment:
This change looks unrelated to changing how the calendar type is determined. If that's correct, I urge you to do such changes in a separate commit for it's easier to debug and revert if the commit only does a single thing.